src/Weaver.php

```
@@ -57,12 +57,12 @@ public function newInstance(string $class, array $args): object
     {
         $aopClass = $this->weave($class);
         $instance = (new ReflectionClass($aopClass))->newInstanceArgs($args);
-        if (! isset($instance->bindings)) {
+        if (! $instance instanceof WeavedInterface) {
             /** @var T $instance  */
             return $instance;
```

PHPStan (2.0.3)

PHPDoc tag @var with type T of object is not subtype of native type mixed.


(varTag.nativeType)